contrib/inventory/puppetdb.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+#
+# Ansible inventory script based on puppetdb.
+#
+# Author: Jan Collijs
+# Source: https://github.com/visibilityspots/ansible-puppet-inventory
+
+import sys
+import os
+import ConfigParser
+import argparse
+from pypuppetdb import connect
+try:
+        import json
+except:
+        import simplejson as json
+
+''' This class will generate a list of nodes based on puppetdb '''
+class PuppetDBInventory():
+    def __init__(self):
+        self.inventory = {}
+        self.read_settings()
+        self.parse_cli_args()
+        if self.args.list:
+            if self.puppetdb_api_version == '4':
```

Comment:
Yes, it will break compatibility with older versions. However, it's also not compatible with newer versions right now, as `connect` doesn't accept an `api_version` kwarg.

You could support both old and new versions if desired, you'd have to detect the version of `pypuppetdb` installed and adjust behavior accordingly. It's probably not a good idea to support only old versions though.